packages/headless/src/controllers/search-parameter-manager/headless-search-parameter-manager.ts

```diff
@@ -232,13 +232,26 @@ function getNumericFacets(state: Partial<SearchParametersState>) {
 }
 
 function getDateFacets(state: Partial<SearchParametersState>) {
-  if (state.dateFacetSet === undefined) {
+  if (state.dateFacetSet === undefined || state.relativeDateSet === undefined) {
     return {};
   }
 
   const df = Object.entries(state.dateFacetSet)
     .map(([facetId, request]) => {
-      const selectedRanges = getSelectedRanges(request.currentValues);
+      const relativeDates = state.relativeDateSet![facetId] || [];
+      const selectedRanges = getSelectedRanges(request.currentValues).map(
+        (facetValue) => {
+          return {
+            ...facetValue,
+            start:
+              relativeDates.find(({value}) => facetValue.start === value) ||
+              facetValue.start,
+            end:
+              relativeDates.find(({value}) => facetValue.end === value) ||
+              facetValue.end,
```

Comment:
Wouldn't the start and end stored in relativeDates be identical to those in the dateFacetSet? If so, then why not simplify to always read `start` and `end` from the dateFacetSet?

The relativeDateSet would be used to fill the relativeStart and relativeEnd,